### ISSUE TITLE
The ES5 build contains Promise.withResolvers() without a polyfill

### ISSUE THREAD

**Issue by stephanrauh** [stephanrauh/ngx-extended-pdf-viewer#2500]
I've noticed that the ES5 files contain calls to `Promise.withResolvers()`. Let's check what went wrong.

Also, the flag `forceUsingLegacyES5` has gone lost. Let's bring it back!

**Comment by AdrianM** [comment]
+1

**Comment by DaanPepping** [comment]
is there a known workaround for this? i updated to 21.3.2 for the csp changes but i'm getting the following error:
<img width="593" alt="image" src="https://github.com/user-attachments/assets/11761900-f88b-4324-a337-721a92001977">

**Comment by stephanrauh** [comment]
@DaanPepping I'd like to know what's going wrong. The error doesn't make sense to me, but it seems the error doesn't care about that. 

Can you open a console window before starting the PDF viewer and execute `Promise.withResolvers`, please? Which output do you see? And what happens if you repeat `Promise.withResolvers` after loading the viewer? Is it a different output?

Other things to check:

- Are you using an old version of zone.js? Please update to the latest possible version. Older versions break `Promise.withResolvers`.
- Which browser are you using? Which version?

**Comment by DaanPepping** [comment]
@stephanrauh thanks for your quick reply, here's what i'm working with:

- executing Promise.withResolvers directly results in an undefined
<img width="152" alt="image" src="https://github.com/user-attachments/assets/aaef0d0d-b6e2-4293-9ee9-d33ad59a2b9e">

software used:
- I'm using arc browser which is chromium based. (127.0.6533.120 (Official Build) (arm64) )
- zone.js 0.13.3
- angular 16.2.12

the viewer never loads because of this so i can't reload it (although i do have 2 csp errors alongside it)  

I'm working on upgrading my app to angular 18 so i'll be able to test it with newer zone.js versions soon, i'll reply here if that resolves the errors i'm having.

**Comment by stephanrauh** [comment]
@DaanPepping Awesome, that sounds like a hot trace. Version 127.0.6533.120 is pretty new. It's hard to believe they don't support `Promise.withResolvers`. But, well, it's hard to argue your screenshot.

So I installed Arc myself. As things go, it installed version 128, so there might be a difference - although I don't think so. When I tried to execute `Promise.withResolvers` in a fresh tab, the result was slightly different:
<img width="395" alt="image" src="https://github.com/user-attachments/assets/31772dbf-103e-4e30-bc0a-a02398a0d97c">

**Comment by stephanrauh** [comment]
Good news: now I'm sure that zone.js 0.13.3 is the culprit. I've managed to reproduce your issue. Now solving it should be easy.
<img width="817" alt="image" src="https://github.com/user-attachments/assets/28603bbc-93a5-494b-a10b-d918097235d8">

**Comment by stephanrauh** [comment]
Please update to version 21.3.3 and tell me if it solves the issue.

If updating is not an option, add and call this method before opening the PDF viewer:

```ts
  /**
   * Angular 16 uses zone.js 0.13.3, and this version has a problem with Promise.withResolvers.
   * If your browser supports Promise.withResolvers, zone.js accidentally overwrites it with "undefined".
   * This method adds a polyfill for Promise.withResolvers if it is not available.
   */
  private polyfillPromiseWithResolversIfZoneJSOverwritesIt() {
    debugger;
    const TypelessPromise = Promise as any;
    if (!TypelessPromise.withResolvers) {
      TypelessPromise.withResolvers = function withResolvers() {
        var a,
          b,
          c = new this(function (resolve, reject) {
            a = resolve;
            b = reject;
          });
        return { resolve: a, reject: b, promise: c };
      };
    }
  }
```

**Comment by stephanrauh** [comment]
The bugfix has landed with version 21.3.3. I'd like to keep this ticket open until someone confirms it solves the issue.

Enjoy!
Stephan

**Comment by stephanrauh** [comment]
That's good news. Hooray!

As for your inline script: try using `[useInlineScripts]="false"`. I tried to implement a CSP detection, but added an attribute just in case. Maybe that was a good idea.

I'm closing the ticket now. Thanks for your help!
Stephan
